File: rplugin/python3/curstr/action/source/directory/path.py
from curstr.action.group import ActionGroup, FileDispatcher
from curstr.action.source.base import Source as Base


class Source(Base):

    DISPATCHER_CLASS = FileDispatcher

    def create(self) -> ActionGroup:
        path = self._cursor.get_file_path()
        absolute_path = self._vim.call("fnamemodify", path, ":p")
        return self._dispatcher.dispatch_one(FileDispatcher.Directory, absolute_path)

File: lua/curstr/action_source/directory/path.lua
local M = {}

function M.create()
  local path = require("curstr.lib.cursor").file_path()
  local abs_path = vim.fn.fnamemodify(path, ":p")
  if not require("curstr.lib.file").is_directory(abs_path) then
    return nil
  end
  return "directory", { path = abs_path }
end

M.description = [[uses a directory path]]

return M
